### PR TITLE
fix(core): more tests and fix for ldml_transforms.cpp

### PR DESCRIPTION
**Pull request by srl295**
For #10734

- fix for bad generation of from/to maps
- 2 fixes for crasher in ldml_transforms.cpp
- There's still an issue in the transform matcher - test case reflects a TODO

@keymanapp-test-bot skip

### CODE REVIEW

File: core/src/ldml/ldml_transforms.cpp [hunk 1]
```diff
@@ -588,7 +588,7 @@ transform_entry::apply(const std::u32string &input, std::u32string &output) cons
     char32_t *s                = new char32_t[group1Len + 1];
     assert(s != nullptr); // TODO-LDML: OOM
     // convert
-    substr.toUTF32((UChar32 *)s, group1Len + 1, status);
+    group1.toUTF32((UChar32 *)s, group1Len + 1, status);
```
Comment (srl295) [inline core/src/ldml/ldml_transforms.cpp:591]: 😬

Comment (srl295) [reply core/src/ldml/ldml_transforms.cpp:591]: i think this assumed the from= only had the matched group and no extra text.

File: developer/src/kmc-ldml/src/compiler/tran.ts [hunk 2]
```diff
@@ -126,7 +126,6 @@ export abstract class TransformCompiler<T extends TransformCompilerType, TranBas
     let cookedFrom = transform.from;
 
     cookedFrom = sections.vars.substituteStrings(cookedFrom, sections);
-    // TODO: handle 'map' case
```
Comment (srl295) [inline developer/src/kmc-ldml/src/compiler/tran.ts:129]: out of date TODO

File: core/src/ldml/ldml_transforms.cpp [hunk 3]
```diff
@@ -749,12 +749,9 @@ transforms::apply(const std::u32string &input, std::u32string &output) {
   for (auto group = transform_groups.begin(); group < transform_groups.end(); group++) {
     // for each transform group
     // break out once there's a match
-    // TODO-LDML: reorders
-    // Assume it's a non reorder group
     /** Length of match within this group*/
 
     // find the first match in this group (if present)
-    // TODO-LDML: check if reorder
```
Comment (srl295) [inline core/src/ldml/ldml_transforms.cpp:757]: out of date todos'!

File: common/web/types/src/ldml-keyboard/pattern-parser.ts [hunk 4]
```diff
@@ -400,7 +400,7 @@ export class VariableParser {
   /**
    * Pattern for matching a capture set reference `($[set])`
    */
-  public static readonly CAPTURE_SET_REFERENCE = /\(\$\[([0-9A-Za-z_]{1,32})\]\)/g;
+  public static readonly CAPTURE_SET_REFERENCE = /\(\$\[([0-9A-Za-z_]{1,32})\]\)/;
```
Comment (mcdurdin) [inline common/web/types/src/ldml-keyboard/pattern-parser.ts:403]: Why the removal of `g`?

**Review by mcdurdin — APPROVED** [review]
LGTM, as far as I can tell, but I don't understand the change to `CAPTURE_SET_REFERENCE`

**Comment by srl295** [comment]
It makes the regex mutable! So it behaves differently on the second use.

File: common/web/types/src/ldml-keyboard/pattern-parser.ts (hunk 4 above)
Comment (srl295) [reply common/web/types/src/ldml-keyboard/pattern-parser.ts:403]: So that using it is idempotent

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.270-beta](https://keyman.com/downloads/releases/beta/17.0.270)

**Comment by mcdurdin** [comment]
> It makes the regex mutable! So it behaves differently on the second use.

Whoa, that's ugly! TIL.

**Pull request merged**